earn/src/components/boost/ImportModal.tsx

```diff
@@ -138,15 +150,43 @@ export default function ImportModal(props: ImportModalProps) {
   const { cardInfo, uniqueId, isOpen, setIsOpen } = props;
   const { activeChain } = useContext(ChainContext);
   const navigate = useNavigate();
+  const provider = useProvider({ chainId: activeChain.id });
 
   const [boostFactor, setBoostFactor] = useState(BOOST_DEFAULT);
+  const [marketInfo, setMarketInfo] = useChainDependentState<MarketInfo | null>(null, activeChain.id);
 
   useEffect(() => {
     if (isOpen) {
       setBoostFactor(BOOST_DEFAULT);
     }
   }, [isOpen]);
 
+  useEffect(() => {
+    let isMounted = true;
+    async function fetchMarketInfo() {
+      if (!provider || !cardInfo) return;
+      const lenderLensContract = new ethers.Contract(
+        ALOE_II_LENDER_LENS_ADDRESS[activeChain.id],
+        KittyLensAbi,
+        provider
+      );
+      const marketInfo = await fetchMarketInfoFor(
+        lenderLensContract,
+        cardInfo.lender0,
+        cardInfo.lender1,
+        cardInfo.token0.decimals,
+        cardInfo.token1.decimals
+      );
+      if (isMounted) {
+        setMarketInfo(marketInfo);
+      }
+    }
+    fetchMarketInfo();
+    return () => {
+      isMounted = false;
+    };
+  }, [activeChain.id, cardInfo, provider, setMarketInfo]);
```

Comment:
This could probably be done with wagmi hooks, but not a big deal either way.

Reply:
I am just re-using old code here.